shared/actions/chat.js

```diff
@@ -0,0 +1,303 @@
+      message.valid &&
+      message.valid.messageBody.messageType === CommonMessageType.text
+    ))
+
+    let snippet
+    try {
+      // $FlowIssue doens't understand try
+      snippet = _makeSnippet(recentMessage.valid.messageBody.text.body, 100)
+    } catch (_) { }
+
+    return new InboxStateRecord({
+      info: convo.info,
+      conversationIDKey: conversationIDToKey(convo.info.id),
+      participants: List(convo.info.writerNames || []) || List(), // TODO in recent order... somehow
+      muted: false,
+      time: 'Time',
+      snippet,
+    })
+  }))
+}
+
+function * _postMessage (action: PostMessage): SagaGenerator<any, any> {
+  const infoSelector = (state: TypedState) => {
+    const convo = state.chat.get('inbox').find(convo => convo.get('conversationIDKey') === action.payload.conversationIDKey)
+    if (convo) {
+      return convo.get('info')
+    }
+    return null
+  }
+
+  const info = yield select(infoSelector)
+
+  if (!info) {
+    console.warn('No info to postmessage!')
+    return
+  }
+
+  const configSelector = (state: TypedState) => {
+    try {
+      return {
+        // $FlowIssue doesn't understand try
+        deviceID: state.config.extendedConfig.device.deviceID,
+        uid: state.config.uid,
+      }
+    } catch (_) {
+      return {
+        deviceID: '',
+        uid: '',
+      }
+    }
+  }
+
+  const {deviceID, uid}: {deviceID: string, uid: string} = ((yield select(configSelector)): any)
+
+  if (!deviceID || !uid) {
+    console.warn('No deviceid/uid to postmessage!')
+    return
+  }
+
+  const clientHeader = {
+    conv: info.triple,
+    tlfName: info.tlfName,
+    tlfPublic: info.visibility === CommonTLFVisibility.public,
+    messageType: CommonMessageType.text,
+    supersedes: 0,
+    sender: Buffer.from(uid, 'hex'),
+    senderDevice: Buffer.from(deviceID, 'hex'),
+  }
+
+  yield call(localPostLocalRpcPromise, {
+    param: {
+      conversationID: keyToConversationID(action.payload.conversationIDKey),
+      msg: {
+        clientHeader,
+        messageBody: {
+          messageType: CommonMessageType.text,
+          text: {
+            body: action.payload.text,
+          },
+        },
+      },
+    },
+  })
+}
+
+function * _incomingMessage (action: IncomingMessage): SagaGenerator<any, any> {
+  if (action.payload.activity.ActivityType === NotifyChatChatActivityType.incomingMessage) {
+    const incomingMessage: ?IncomingMessageRPCType = action.payload.activity.IncomingMessage
+    if (incomingMessage) {
+      const messageUnboxed: MessageUnboxed = incomingMessage.message
+      const yourNameSelector = (state: TypedState) => state.config.username
+      const yourName = yield select(yourNameSelector)
+      const message = _threadToStorable(messageUnboxed, 0, yourName)
+
+      yield put({
+        type: Constants.appendMessages,
+        payload: {
+          conversationIDKey: conversationIDToKey(incomingMessage.convID),
+          messages: [message],
```

Comment:
I think in general we don't want to log message contents. We really don't want to see messages in debug logs.